site/app/templates/course/CourseMaterials.twig

```diff
@@ -80,18 +80,18 @@
                         {% set url_title = "" %}
                         {% set url_url = "" %}
                     {% endif %}
-                    <a class="key_to_click" tabindex="0" onclick='newEditCourseMaterialsForm("{{ course_material.getId }}", "{{ course_material.getPriority }}", {{course_material.getSections|map(s => s.getSectionId)|json_encode|raw}}, "{{course_material.isHiddenFromStudents}}", "{{ course_material.getReleaseDate|date(date_format) }}", "{{ course_material.isLink }}", "{{ url_title }}", "{{ url_url }}")'> <i class="fas fa-pencil-alt black-btn" aria-hidden="true" style="font-size: 16px; margin: 5px;"></i></a>
+                    <a class="key_to_click" tabindex="0" onclick='newEditCourseMaterialsForm(this)' data-id="{{ course_material.getId }}" data-priority="{{ course_material.getPriority }}" data-sections={{course_material.getSections|map(s => s.getSectionId)|json_encode|raw}} data-hidden-state="{{course_material.isHiddenFromStudents}}" data-release-time="{{ course_material.getReleaseDate|date(date_format) }}" data-is-link="{{ course_material.isLink }}" data-link-title="{{ url_title }}" data-link-url="{{ url_url }}"> <i class="fas fa-pencil-alt black-btn" aria-hidden="true" style="font-size: 16px; margin: 5px;"></i></a>
```

Comment:
This line has been broken down as suggested.

Thanks for testing thoroughly, the alert you faced is intentional but it shouldn't have been thrown, so, it was indeed an issue. It was happening because partial sections weren't getting recognized when applying recursive updates, so that alert said that you should select 'No' radio button when asked about whether you want to restrict the folder to some sections. Please see if you can reproduce that issue now, the alert message has also been tweaked. A couple of Cypress tests have been added. Also, support for partial 'Hide from Students' setting in folder edit form wasn't available earlier, it has been implemented too.